src/Mafia/Cabal/Package.hs

```diff
@@ -211,33 +214,28 @@ getPackageId dir = do
     Left _ ->
       return Nothing
     Right f -> do
-      mpid <- readPackageId f
-      case mpid of
-        Nothing  -> left (CabalCouldNotReadPackageId f)
-        Just pid -> return (Just pid)
-
-readPackageId :: MonadIO m => File -> m (Maybe PackageId)
-readPackageId cabalFile = do
-  text <- fromMaybe T.empty `liftM` readUtf8 cabalFile
-
-  let findName    = fmap mkPackageName . findField "name"
-      findVersion = (parseVersion =<<) . findField "version"
-
-  let lines   = fmap T.words (T.lines text)
-      name    = listToMaybe . mapMaybe findName    $ lines
-      version = listToMaybe . mapMaybe findVersion $ lines
-
-  return (PackageId <$> name <*> version)
-
-findField :: Text -> [Text] -> Maybe Text
-findField field = \case
-  (key:value:_)
-    | T.toLower (field <> ":") == T.toLower key ->
-    Just value
-  (_:_) ->
-    Nothing
-  [] ->
-    Nothing
+      liftM Just $ readPackageId f
+
+readPackageId :: MonadIO m => File -> EitherT CabalError m PackageId
+readPackageId file = do
+  msrc <- liftM T.unpack `liftM` readUtf8 file
+  case msrc of
+    Nothing ->
+      left $ CabalCouldNotReadPackageId file
+    Just src ->
+      case parsePackageDescription src of
+        ParseFailed err ->
+          left $ CabalFileParseError file err
```

Comment:
As an added bonus, we now get a specific parse error from Cabal, instead of just "could not read package-id"